resources/migrations/001_update_migrations.yaml

```diff
@@ -8253,6 +8253,152 @@ databaseChangeLog:
+              UPDATE report_dashboard
+              SET last_viewed_at = COALESCE(GREATEST(
+                (
+                  SELECT MAX(timestamp)
+                  FROM view_log
+                  WHERE report_dashboard.id = model_id
+                    AND model = 'dashboard'
+                ),
+                (
+                   SELECT
+                     MAX(last_used_at) AS last_card_used_at
+                   FROM report_card
+                   JOIN report_dashboardcard
+                     ON report_dashboardcard.card_id = report_card.id
+                   WHERE report_dashboardcard.dashboard_id = report_dashboard.id
+                )),
+                NOW())
+        - sql:
+            dbms: postgresql
+            sql: >-
+              WITH from_view_log AS (
+                  SELECT model_id AS dashboard_id,
+                         MAX(timestamp) AS last_viewed_at
+                  FROM view_log
+                  WHERE model = 'dashboard'
+                  GROUP BY model_id
+              )
+              UPDATE report_dashboard
+              SET last_viewed_at = from_view_log.last_viewed_at
+              FROM from_view_log
+              WHERE from_view_log.dashboard_id = report_dashboard.id;
+
+              WITH card_last_viewed_at AS (
+                   SELECT
+                     report_dashboardcard.dashboard_id AS dashboard_id,
+                     MAX(last_used_at) AS last_card_used_at
+                   FROM report_card
+                   JOIN report_dashboardcard
+                     ON report_dashboardcard.card_id = report_card.id
+                   GROUP BY report_dashboardcard.dashboard_id
+              )
+              UPDATE report_dashboard
+              SET last_viewed_at = card_last_viewed_at.last_card_used_at
+              FROM card_last_viewed_at
+              WHERE report_dashboard.id = card_last_viewed_at.dashboard_id
+                AND (
+                  report_dashboard.last_viewed_at IS NULL
+                  OR report_dashboard.last_viewed_at < card_last_viewed_at.last_card_used_at
+                );
+
+              UPDATE report_dashboard SET last_viewed_at = NOW() WHERE last_viewed_at IS NULL;
+
+  - changeSet:
+      id: v51.2024-06-14T09:15:36
+      author: johnswanson
+      comment: >-
+        Add NOT NULL constraint to report_dashboard.last_viewed_at
+      changes:
+        - addNotNullConstraint:
+            tableName: report_dashboard
+            columnName: last_viewed_at
+            columnDataType: ${timestamp_type}
+            defaultValueComputed: current_timestamp
+      rollback:
+        - dropNotNullConstraint:
+            tableName: report_dashboard
+            columnName: last_viewed_at
+            columnDataType: ${timestamp_type}
+
+
+  - changeSet:
+      id: v51.2024-06-18T12:13:03
+      author: johnswanson
+      comment: Add default value to report_dashboard.last_viewed_at
+      changes:
+        - addDefaultValue:
```

Comment:
Are we sure we want to make this assumption? what if someone creates a dashboard pragmatically? how about serialization? how about dashboard duplication?

Reply:
To be clear, this isn't intended as an audit event or anything like that. It's just a marker intended as a way to find old, stale dashboards. I think defaulting to `now()` makes sense given that - though you do make a good point regarding serialization, I think in an ideal world we would export the `last_viewed_at` of dashboards and the `last_used_at` of cards, so that you could still identify stale items after an import. 🤔

Reply:
> so that you could still identify stale items after an import. 

The need to identify stale items after an import could be a genuine use case. Maybe during serialization import we should compare `last_viewed_at` from both the imported entity and the existing entity, and keep the maximum.

Reply:
Checked with arakaki and confirmed `now` makes sense [here](https://metaboat.slack.com/archives/C0641E4PB9B/p1720549646851889). Would be great to leave a comment explaining why